test/e2e/util.go

```diff
@@ -194,6 +196,9 @@ func routeExternalDNS(name, zoneID, zoneDomain string) operatorv1alpha1.External
 					AnnotationFilter: map[string]string{
 						"external-dns.mydomain.org/publish": "yes",
 					},
+					OpenShiftRoute: &operatorv1alpha1.ExternalDNSOpenShiftRouteOptions{
+						RouterName: routerName,
+					},
```

Comment:
This will make the existing Mustafa's test not working as ExteranlDNS > 0.10.1 will be needed for his case too.
Either you make the route addition optional if `routerName` is empty, or both cases need to be merged.